### PR TITLE
NIFI-10693 Remove direct proxy configuration properties from PutBigQuery

### PR DESCRIPTION
**Pull request by bejancsaba**
# Summary

[NIFI-10693](https://issues.apache.org/jira/browse/NIFI-10693)

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [x] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [x] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [x] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [x] Pull Request based on current revision of the `main` branch
- [x] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [x] Build completed using `mvn clean install -P contrib-check`
  - [x] JDK 8
  - [ ] JDK 11
  - [ ] JDK 17

### Licensing

- [ ] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [ ] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [ ] Documentation formatting appears as expected in rendered files

### CODE REVIEW

**Review by exceptionfactory — COMMENTED** [review]
Thanks for the improvement @bejancsaba.

How will this impact existing deployments using this Processor?

The `InvokeHTTP` Processor supports both direct Proxy configuration and Proxy Configuration Service properties for compatibility. Recent changes added deprecation logging warnings for use of the direct properties, so it seems like that might be necessary for now to avoid breaking flows that use the direct properties.

**Comment by turcsanyip** [comment]
@exceptionfactory Newly added processors should only support the Proxy (or SSL) controller service and not the direct configuration properties.
`PutBigQuery` was just added in 1.18.0 so in my opinion it may be acceptable to remove those properties with mentioning it in the Migration Guidance.

**Comment by exceptionfactory** [comment]
Thanks for the background on release timing @turcsanyip.

I agree that new Processors should only use the Proxy Configuration Service. Since this Processor was just released in 1.18.0, and because the individual Proxy properties are optional, it seems reasonable to move forward with removing the individual properties and adding a note to Migration Guidance for subsequent release versions.

It looks like there are automated test failures related to the property changes that need to be addressed, but the general plan sounds good in light of that background.

**Comment by turcsanyip** [comment]
Thanks for your answer @exceptionfactory!

Actually, while I was testing the processor I noticed that the proxy is not used at all (configured either via the legacy properties or the controller service). So at least we will not break anything working.

It seems it is not enough to add the proxy config for the HTTP transport, but we need to add it for GRPC too. It can be done via `BigQueryWriteSettings.Builder.setTransportChannelProvider()` but the config is not straightforward.

@bejancsaba I think we can either fix the proxy here or re-scope this PR, remove the proxy config at all and add a follow-up jira for fixing it.

**Comment by bejancsaba** [comment]
> @bejancsaba I think we can either fix the proxy here or re-scope this PR, remove the proxy config at all and add a follow-up jira for fixing it.

Removed the proxy configuration for now and created a dedicated Jira for the Proxy work: https://issues.apache.org/jira/browse/NIFI-10707

File: nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/bigquery/PutBigQuery.java [hunk 1]
```diff
@@ -119,6 +127,15 @@ public class PutBigQuery extends AbstractBigQueryProcessor {
...
     private boolean skipInvalidRows;
 
+    public static final PropertyDescriptor PROJECT_ID = new PropertyDescriptor.Builder()
+        .name(PROJECT_ID_NAME)
+        .displayName("Project ID")
+        .description(PROJECT_ID_DESC)
+        .required(true)
+        .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+        .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+        .build();
```
[the remark this answers is not in the document]
Comment (bejancsaba) [reply nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/bigquery/PutBigQuery.java:137]: Thanks, applied I was not aware of this neat functionality :)

**Review by turcsanyip — APPROVED** [review]
Thanks @bejancsaba and @exceptionfactory!

+1 LGTM
Mergng to main.

**Pull request closed without merging**